### PR TITLE
[KEYCLOAK-4902] - Refactoring and improvements to processing of authz requests

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Review by mhajas — DISMISSED** [review]
I have just two minor things. If you don't find it important it is ready to merge from my POV.

**Comment by pedroigor** [comment]
@mhajas, done.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/authz/ConflictingScopePermissionTest.java [hunk 1]
```diff
@@ -111,6 +164,59 @@ public void testMartaCanAccessResourceAWithExecuteAndWrite() {
...
+                    break;
+                case "Resource C":
+                    assertEquals(3, permission.getScopes().size());
+                    assertTrue(permission.getScopes().contains("execute"));
+                    assertTrue(permission.getScopes().contains("write"));
+                    assertTrue(permission.getScopes().contains("read"));
+                    permissions.remove(permission);
+                    break;
+                case "Resource B":
+                    assertEquals(3, permission.getScopes().size());
```
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/authz/ConflictingScopePermissionTest.java]: This and three following lines can be replaced by `assertThat(permission.getScopes(), containsInAnyOrder("execute", "write", "read"));` - improves message on failure.

File: model/jpa/src/main/java/org/keycloak/authorization/jpa/store/JPAPolicyStore.java [hunk 2]
```diff
@@ -269,14 +291,13 @@ public Policy findByName(String name, String resourceServerId) {
         query.setParameter("serverId", resourceServerId);
 
         List<String> result = query.getResultList();
-        List<Policy> list = new LinkedList<>();
+
         for (String id : result) {
             Policy policy = provider.getStoreFactory().getPolicyStore().findById(id, resourceServerId);
             if (Objects.nonNull(policy)) {
-                list.add(policy);
+                consumer.accept(policy);
```
Comment (hmlnarik) [inline model/jpa/src/main/java/org/keycloak/authorization/jpa/store/JPAPolicyStore.java]: `for` cycle could be replaced by
```java
result.stream()
  .map(id -> provider.getStoreFactory().getPolicyStore().findById(id, resourceServerId))
  .filter(Objects::nonNull)
  .forEach(consumer::accept);
```
which would prepare it to [JPA streams](https://vladmihalcea.com/whats-new-in-jpa-2-2-stream-the-result-of-a-query-execution/)

**Review by mhajas — DISMISSED** [review]

**Review by hmlnarik — APPROVED** [review]
Approving (mostly based on @mhajas's approval)

**Pull request merged**